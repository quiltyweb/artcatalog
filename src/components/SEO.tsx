import * as React from "react";

type SEOProps = {
  pageTitle: string;
  description?: string;
  siteTitle?: string;
  image?: string;
  children?: React.ReactNode;
};

const SEO: React.FunctionComponent<SEOProps> = ({
  pageTitle,
  description = "Brushella Art and Decor store",
  siteTitle = "Brushella",
  image = "/brushella-icon.svg",
  children,
}): React.ReactElement => {
  return (
    <>
      <html lang="en" />
      <title>
        {pageTitle} {siteTitle}
      </title>
      <meta name="description" content={description} />
      <meta name="image" content={image} />
      <link rel="icon" href={image} type="image/svg+xml" sizes="any" />
      <link
        rel="stylesheet"
        type="text/css"
        charSet="UTF-8"
        href="https://cdnjs.cloudflare.com/ajax/libs/slick-carousel/1.6.0/slick.min.css"
      />
      <link
        rel="stylesheet"
        type="text/css"
        href="https://cdnjs.cloudflare.com/ajax/libs/slick-carousel/1.6.0/slick-theme.min.css"
      />
      {children}
    </>
  );
};

export default SEO;
