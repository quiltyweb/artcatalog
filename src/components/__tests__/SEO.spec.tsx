import React from 'react';
import { render, screen } from '@testing-library/react';
import SEO from '../SEO';

beforeEach(() => {
  jest.clearAllMocks();
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('SEO', () => {
  it('render childrens correctly', async () => {
    render(
        <SEO>
          <div>Text for SEO</div>
        </SEO>
    );
    screen.getByText('Text for SEO');
  });
});
