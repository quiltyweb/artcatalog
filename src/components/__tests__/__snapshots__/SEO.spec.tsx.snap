// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO render children correctly 1`] = `
<DocumentFragment>
  <link
    href="test-file-stub"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="test-file-stub"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="test-file-stub"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    color="#000000"
    href="test-file-stub"
    rel="mask-icon"
  />
  <link
    href="test-file-stub"
    rel="shortcut icon"
  />
  <meta
    content="#ffffff"
    name="msapplication-TileColor"
  />
  <meta
    content="../images/browserconfig.xml"
    name="msapplication-config"
  />
  <meta
    content="#ffffff"
    name="theme-color"
  />
  <link
    charset="UTF-8"
    href="https://cdnjs.cloudflare.com/ajax/libs/slick-carousel/1.6.0/slick.min.css"
    rel="stylesheet"
    type="text/css"
  />
  <link
    href="https://cdnjs.cloudflare.com/ajax/libs/slick-carousel/1.6.0/slick-theme.min.css"
    rel="stylesheet"
    type="text/css"
  />
  <title
    id="title"
  >
    www.brushella.art - Welcome
  </title>
  <meta
    content="Brushella store offers original paintings, art decor, crafts, clothing and more"
    id="description"
    name="description"
  />
  <div>
    Text for SEO
  </div>
</DocumentFragment>
`;
